Add "Access Graph" docs (#35154)
gravitational/teleport: docs/pages/access-graph.mdx

---
title: Teleport Access Graph
description: A reference for Teleport Access Graph.
---

<Admonition type="tip" title="Preview">

>   Teleport Access Graph is currently in Preview. Currently, it is only available
>   to the Teleport Cloud Team users. Currently, only SSH server are supported.

Reviewer: ```suggestion
  Teleport Access Graph is currently in Preview. Currently, it is only available
  to Teleport Team users, and only visualizes access to SSH servers enrolled with your Teleport clusters.
```
Suggested to:
- Make this consistent with the rest of the docs, which say "Teleport Team", rather than "Teleport Cloud Team"
- Be more explicit about what "are supported" means. I'm not sure if my suggestion is technically correct, though.

    Actions are the things that users can or cannot do. Actions are related to
    resources. For example, a user can SSH into a node.

5. Deny Actions

    ![Deny Action Node](../img/access-graph/deny-action-node.png)

    Deny Actions are the things that users cannot do. Deny Actions are related to
    resources. For example, a user cannot SSH into a node.

6. Temporary Actions

    ![Temporary Action Node](../img/access-graph/temporary-action-node.png)

    Temporary Actions are created when a user is granted temporary access to a
    resource. They are automatically deleted when the user's access expires.

7. Resource Groups

    ![Resource Group Node](../img/access-graph/resource-group-node.png)

    Resource Groups are collections of resources. They can be used to organize
    resources based on their role or team.

8. Resources

    ![Resource Node](../img/access-graph/resource-node.png)

    Resources are the things that users can or cannot access. They can be
    servers, databases, or Kubernetes clusters.

## Graph paths

Teleport Access Graph shows the relationships between users, roles, and
resources. It does this by showing paths between nodes. Paths are the

> relationships between nodes. Paths always go from:
> 
>   Users to User Groups to Actions to Resource Groups to Resources

Reviewer: ```suggestion
relationships between nodes. Paths always connect nodes in the following order:

1. Users
1. User Groups
1. Actions
1. Resource Groups
1. Resources
```
Suggested to make this a little easier to read

to your Teleport resources, the Access Graph will reflect those changes.

#### Users

Users are created from Teleport Users.

#### User Groups

User Groups are created from Teleport Roles and access requests. Roles create a User Groups where the members
are the users that have that role. Access requests create a temporary User Group where the members are the users that
got the access through the accepted access request.

#### Actions

Actions are created from Teleport Roles. Actions can be divided into three
categories:

1. Allow Actions

    Allow Actions are created from Teleport Roles. Allow Actions are the things
    that users can do. For example, a user can SSH into a node.

2. Deny Actions

    Deny Actions are created from Teleport Roles. Deny Actions are the things
    that users cannot do. For example, a user cannot SSH into a node. Deny
    Actions take precedence over Allow Actions.

3. Temporary Actions

    Temporary Actions are created when a user is granted temporary access to a
    resource. They are automatically deleted when the user's access expires.

#### Resource Groups

> Resource Groups are created from Teleport Roles.

Reviewer: ```suggestion
Resource Groups are created from Teleport roles.
```

#### Resources

Resources are created from Teleport resources like nodes, databases, and

> Kubernetes clusters.

Reviewer: ```suggestion
Kubernetes clusters. Currently, only SSH nodes are supported.
```
Is this what the Admonition at the top means? If so, I've suggested mentioning it here as well.

---
title: Teleport Access Graph
description: A reference for Teleport Access Graph.
---

<Admonition type="tip" title="Preview">
  Teleport Access Graph is currently in Preview. Currently, it is only available
  to Teleport Team users, and only visualizes access to SSH servers enrolled with your Teleport clusters.
</Admonition>

Teleport Access Graph is a new way to visualize and understand access to your
infrastructure. It provides a visual representation of the relationships between
users, roles, and resources in your organization. It can help you answer
questions like:

- What resources can a specific user access?
- What users can access a specific resource?
- What are the relationships between users, roles, and resources?

After logging into the Teleport UI, go to the Management tab. If enabled, Access Graph options can be found
under the Permission Management section.

![Teleport Access Graph](../img/access-graph/main-view.png)

## Graph nodes

> Teleport Access Graph divides your infrastructure into five main components:

Reviewer 2: > five main components

There's 8 below from what I can see.

---
title: Teleport Access Graph
description: A reference for Teleport Access Graph.
---

<Admonition type="tip" title="Preview">
  Teleport Access Graph is currently in Preview. Currently, it is only available

>   to the Teleport Cloud Team users. Not all Teleport resources are currently

Reviewer: Does "Teleport Cloud Team" mean both "Teleport Enterprise Cloud" and "Teleport Team"?

>   supported.

Reviewer: Rather than saying that not all resources are supported, we should list the resources that are supported.
